src/ios/GeolocationPlugin.m

```diff
@@ -0,0 +1,314 @@
+//
+//  GeolocationPlugin.m
+//  Geolocation
+//
+//  Created by dev on 2/9/16.
+//
+//
+
+#import "GeolocationPlugin.h"
+
+@implementation GeolocationPlugin
+
+-(void)start:(CDVInvokedUrlCommand *)command
+{
+    // NSString *callbackId = command.callbackId;
+    //  [self returnLocationInfo:callbackId andKeepCallback:"OKay"];
+    
+    CDVPluginResult* pluginResult = nil;
+    coord = [command.arguments objectAtIndex:0];
+    
+    
+    //[self InitData];
+    
+    if (![self.thread isExecuting]) {
+        NSLog(@"Starting thread...");
+        loopflag = TRUE;
+        [self.thread start];
+    }
+}
+
+-(NSThread*)thread{
+    if (!_thread) {
+        _thread = [[NSThread alloc]initWithTarget:self selector:@selector(longloop) object:nil];
+    }
+    return _thread;
+}
+-(void)longloop{
+    while (loopflag) {
+        [self performSelector:@selector(InitData) onThread:[NSThread mainThread] withObject:[NSNumber numberWithLong:3] waitUntilDone:YES];
+        sleep(75);
+    }
+}
+
+
+-(void)stop:(CDVInvokedUrlCommand *)command
+{
+    loopflag = FALSE;
+    if ([self.thread isExecuting]) {
+        NSLog(@"Stopping thread...");
```

Comment:
this won't stop location or region monitoring, see [this guide](https://developer.apple.com/library/ios/documentation/UserExperience/Conceptual/LocationAwarenessPG/CoreLocation/CoreLocation.html), search for stopUpdating